### DISCUSSION TITLE
How to get filters to work with TextLintEngine when used as a module?

### DISCUSSION THREAD

**Discussion by Piedone** [textlint/textlint#1389] [Q&A] [answered]
I'm trying to run textlint v13.3.3 with textlint-filter-rule-comments v 1.2.2 as a module like this:

```js
const engine = new textlint.TextLintEngine({
    rules: [
        'doubled-spaces',
    ],
    filters: {
        comments: true,
    },
});

// Reading file to fileContent.

engine.executeOnText(fileContent, '.md');
```

The rule is correctly applied, but the filter doesn't seem so, because e.g. the below Markdown still fails on it with "warning doubled-spaces: Found doubled spaces."

```
<!-- textlint-disable -->

> [!NOTE]
> The code samples in the documentation reference the latest versions of the workflows and actions from the...

<!-- textlint-enable -->
```

What am I doing wrong, how can the filter be loaded and executed? My full code with this change is here: https://github.com/Lombiq/NodeJs-Extensions/commit/cf64bf0766d831df963586bf4f70df48b6a6d91c

I tried this too, but it doesn't help:

```js
const engine = new textlint.TextLintEngine({
    rules: [
        'doubled-spaces',
    ],
    filters: [
        'comments',
    ],
});
```

It's as if the filter isn't applied at all, but I can't get any more information.

Thank you!

**Comment by azu** [comment, marked as answer]
This is undocument way.
Probabaly, you can use `filterRules` instead of `filters`.
https://github.com/textlint/textlint/blob/67b62c6e1d867cdfa223e43c440dfd4116eb8868/packages/textlint/src/config/config.ts#L361-L363

⚠️ `TextLintEngine` is deprecated in [textlint v14.0.0 · textlint](https://textlint.github.io/blog/2024/02/03/textlint-14).

**Reply by Piedone** [reply]
Thank you for your quick reply and detailed info! And indeed, just renaming `filters` to `filterRules` works, awesome, thanks!

We'll update to v14 eventually, but for now, I'd like to tackle just filters.
